src/index.ts

```diff
@@ -0,0 +1,242 @@
+import p from 'path'
+import fs from 'fs'
+import type babelCore from '@babel/core'
+import type {Primitive} from 'type-fest'
+import {getReplacement, requireFromString} from './helpers'
+
+type VisitorState = {
+  file: {
+    opts: babelCore.TransformOptions
+  }
+}
+
+export default function prevalPlugin(
+  babel: typeof babelCore,
+): babelCore.PluginObj<VisitorState> {
+  const {types: t, template, transformFromAst} = babel
+  const assignmentBuilder = template('const NAME = VALUE')
+  return {
+    name: 'preval',
+    visitor: {
+      Program(path, {file: {opts: fileOpts}}) {
+        const firstNode = path.node.body[0] || {}
+        const comments = firstNode.leadingComments ?? []
+        const prevalComment = comments.find(isPrevalComment)
+
+        if (!prevalComment) {
+          return
+        }
+
+        prevalComment.value = ' this file was prevaled'
+
+        // @ts-expect-error the types for this is wrong...
+        const {code: string}: {code: string} = transformFromAst(
+          path.node,
+          // @ts-expect-error the types for this is wrong...
+          null,
+          /* istanbul ignore next (babel 6 vs babel 7 check) */
+          babel.version.startsWith('6.')
+            ? {}
+            : {
+                babelrc: false,
+                configFile: false,
+              },
+        )
+
+        const replacement = getReplacement({string, fileOpts, babel})
+
+        // istanbul ignore next because this should never happen, but TypeScript needs me to handle it
+        if (!replacement) return
```

Comment:
Throughout this migration a return like this is used to narrow the type of replacement for the next line. We don't believe it will ever actually happen, and none of the existing test cases hit it (hence the ignore coverage comment), but just highlighting a functional difference from the old version.

I will also mention that `babel-plugin-codegen` appears to be [handling these cases slightly differently](https://github.com/kentcdodds/babel-plugin-codegen/blob/main/src/helpers.ts#L83), however this was existing prior to the TS migration.

Reply:
I'm not super concerned as long as it's working and nobody's complaining. I don't need them to be consistent with each other.